filesmanager/filesmanager.py

```diff
@@ -131,39 +160,103 @@ def student_view(self, context=None):
+        if not js_content:
+            js_content_parsed = (
+                f"function {js_main_function}(runtime, element, context) {{"
+                "console.error('Something went wrong with XBlock rendering');"
+                "}")
+
         # Add i18n js
-        # statici18n_js_url = self._get_statici18n_js_url()
-        # if statici18n_js_url:
-            #frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
+        statici18n_js_url = self._get_statici18n_js_url()
+        if statici18n_js_url:
+           frag.add_javascript_url(self.runtime.local_resource_url(self, statici18n_js_url))
 
-        # Adding the correct route of the bundle
-        frag.add_javascript(self.resource_string("static/html/bundle.js"))
-        frag.initialize_js('FilesManagerXBlock')
+        js_context  = {
+           "xblock_id": self.block_id,
+            "is_edit_view": False
+        }
 
+        frag.add_javascript(js_content_parsed)
+        frag.initialize_js(js_main_function, json_args=js_context)
         return frag
 
     def studio_view(self, context=None):
         """
         The edit view of the FilesManagerXBlock in Studio.
         """
-        html = self.resource_string("static/html/filesmanager.html")
-        frag = Fragment(html.format(self=self))
+        if context:
+           pass  # TO-DO: do something based on the context.
+
+        # Main function name for the XBlock
+        js_xblock_function = f"XBlockMainEdit{self.block_id_parsed}"
+        react_app_root_id = f"files-manager-app-root-{self.block_id_parsed}-edit"
+
+        # Read the JavaScript content from the bundle file
+        js_content = self.read_file("static/html/bundle.js")
+        if js_content:
+            # Replace the default React app root with the new one and update the main function name
+            new_content = js_content.replace('files-manager-app-root', react_app_root_id)
+            js_content = new_content.replace('FilesManagerXBlock', js_xblock_function)
+
+        # Create the HTML fragment with the React app and JavaScript
+        html = f"<div id='{react_app_root_id}'></div><script defer='defer'>{js_content}</script>"
+        frag = Fragment(html)
         frag.add_css(self.resource_string("static/css/filesmanager.css"))
 
+        # Define the main function for the XBlock
+        js_main_function = f"Main_{js_xblock_function}"
+
+        # Define the parsed JavaScript content
+        js_content_parsed = (
+           f"function {js_main_function}(runtime, element, context) {{"
+           f"{js_xblock_function}(runtime, element, context);"
+           "}")
+
+        # Handle the case where there's an error getting the bundle file
+        if not js_content:
+            js_content_parsed = (
+                f"function {js_main_function}(runtime, element, context) {{"
+                "console.error('Something went wrong with XBlock rendering');"
+                "}")
```

Comment:
```suggestion
        # Define the parsed JavaScript content
        js_content_parsed = (
           f"function {js_main_function}(runtime, element, context) {{"
           f"{js_xblock_function}(runtime, element, context);"
        "}")

        # Handle the case where there's an error getting the bundle file
        if not js_content:
            js_content_parsed = (
                f"function {js_main_function}(runtime, element, context) {{"
                "console.error('Something went wrong with XBlock rendering');"
            "}")
```

Reply:
We also shouldn't use inline comments to describe what we're doing so the code should be enough. Could we remove them?

Reply:
This is a pattern we try to avoid. Clean code should be self-explanatory, and in cases where the code itself is not enough we use inline comments

Reply:
Exactly ^; we usually try to use inline comments to explain _why_ we're doing something so it's easier for whoever's reading the code. But we should avoid descriptive inline comments.